utils/python/CIME/utils.py

```diff
@@ -595,42 +595,57 @@ def convert_to_type(value, type_str, vid=""):
             try:
                 value = int(value)
             except ValueError:
-                expect(False, "Entry %s was listed as type int but value '%s' is not valid int" % (vid, value))
+                if not ok_to_fail:
+                    expect(False, "Entry %s was listed as type int but value '%s' is not valid int" % (vid, value))
 
         elif type_str == "logical":
-            expect(value in ["TRUE", "FALSE","true","false"],
-                   "Entry %s was listed as type logical but had val '%s' instead of TRUE or FALSE" % (vid, value))
-            value = value == "TRUE" or value == "true"
+            if ok_to_fail:
+                pass
+            else:
+                expect(value in ["TRUE", "FALSE","true","false"],
+                       "Entry %s was listed as type logical but had val '%s' instead of TRUE or FALSE" % (vid, value))
+                value = value == "TRUE" or value == "true"
 
         elif type_str == "real":
             try:
                 value = float(value)
             except:
-                expect(False, "Entry %s was listed as type real but value '%s' is not valid real" % (vid, value))
+                if not ok_to_fail:
+                    expect(False, "Entry %s was listed as type real but value '%s' is not valid real" % (vid, value))
 
         else:
             expect(False, "Unknown type '%s'" % type_str)
 
     return value
 
-def convert_to_string(value, type_str, vid=""):
+def convert_to_string(value, type_str=None, vid=""):
```

Comment:
Is anyone using type_str=None? I don't see how that's needed for this change. Also, if we're just going to return value as-is if type_str is None, why even call the function in the first place?